DQMOffline/Trigger/plugins/PhotonMonitor.h

```diff
@@ -0,0 +1,121 @@
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronFwd.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonFwd.h"
+
+class GenericTriggerEventFlag;
+
+struct MEbinning {
+  int nbins;
+  double xmin;
+  double xmax;
+};
+
+struct PhotonME {
+  MonitorElement* numerator;
+  MonitorElement* denominator;
+};
+//
+// class declaration
+//
+
+class PhotonMonitor : public DQMEDAnalyzer 
+{
+public:
+  PhotonMonitor( const edm::ParameterSet& );
+  ~PhotonMonitor();
+  static void fillDescriptions(edm::ConfigurationDescriptions & descriptions);
+  static void fillHistoPSetDescription(edm::ParameterSetDescription & pset);
+  static void fillHistoLSPSetDescription(edm::ParameterSetDescription & pset);
+
+protected:
+
+  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+  void bookME(DQMStore::IBooker &, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbins, double xmin, double xmax);
+  void bookME(DQMStore::IBooker &, PhotonME& me, const std::string& histname, const std::string& histtitle, const std::vector<double>& binningX);
+  void bookME(DQMStore::IBooker &, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, double ymin, double ymax);
+  void bookME(DQMStore::IBooker &, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax);
+  void bookME(DQMStore::IBooker &, PhotonME& me, const std::string& histname, const std::string& histtitle, const std::vector<double>& binningX, const std::vector<double>& binningY);
+  void setTitle(PhotonME& me, std::string titleX, std::string titleY);
+
+  void analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup) override;
+
+private:
+  static MEbinning getHistoPSet    (edm::ParameterSet pset);
+  static MEbinning getHistoLSPSet  (edm::ParameterSet pset);
+
+  std::string folderName_;
+  std::string histoSuffix_;
+
+  edm::EDGetTokenT<reco::PFMETCollection>       metToken_;
+  edm::EDGetTokenT<reco::PFJetCollection>       jetToken_;
+  edm::EDGetTokenT<reco::GsfElectronCollection> eleToken_;
+  edm::EDGetTokenT<reco::PhotonCollection>     photonToken_;
+
+
+
+
+  std::vector<double> photon_variable_binning_;
+  MEbinning           photon_binning_;
+  MEbinning           ls_binning_;
+
+  PhotonME photonME_;
+  PhotonME photonEtaME_;
+  PhotonME photonPhiME_;
+  PhotonME photonME_variableBinning_;
+  PhotonME photonVsLS_;
+  PhotonME photonEtaPhiME_;   
+  PhotonME photonr9ME_;   
+  PhotonME photonHoverEME_;   
+
+
+
+  GenericTriggerEventFlag* num_genTriggerEventFlag_;
+  GenericTriggerEventFlag* den_genTriggerEventFlag_;
+
+  StringCutObjectSelector<reco::MET,true>         metSelection_;
```

Comment:
do you expect significantly more complex selections than those in the fillDescriptions? If not, the StringCutObjectSelector is very slow and memory intensive for your needs